py/worker/binary/optimizer_worker.py

```diff
@@ -0,0 +1,125 @@
+# limitations under the License.
+"""Generic worker which propose to worklist_scheduler optimizer."""
+
+import asyncio
+from typing import Tuple, Dict
+import warnings
+
+import os
+from absl import app
+from absl import flags
+from sight import sight
+from sight.sight import Sight
+from sight.widgets.decision import proposal
+from sight.widgets.decision.optimizers.bayes_opt_client import BayesOptOptimizerClient
+
+
+def warn(*args, **kwargs):
+  pass
+
+
+warnings.warn = warn
+
+FLAGS = flags.FLAGS
+
+
+def get_question_label_to_propose_actions():
+  return "Fvs"
+
+
+def get_question_label():
+  return "Generic"
+
+
+# def main(sight: Sight, action: Dict[str, int], is_last_action: bool) -> Tuple[float, Dict[str, int]]:
+
+#   # using actions we received from optimizer to propose actions to
+#   # worklist_scheduler
+#   outcome = asyncio.run(
+#       proposal.propose_actions(sight,
+#                                get_question_label_to_propose_actions(),
+#                                action_dict=action,
+#                                is_last_action=is_last_action))
+
+#   vals = list(action.values())
+#   # some mechanchism to calculate reward from the response of WS worker
+#   reward = sum(xi**2 for xi in vals)
+
+#   return reward, outcome
+
+
+async def propose_parallely(sight: Sight, opt_obj):
+  num_questions = int(os.getenv('NUM_QUESTIONS', '1'))
+  batch_size = 5
+  is_last_action = False
+  # tasks = []
+  actions = []
+  rewards = []
+  outcomes = []
+
+  for i in range(0, num_questions, batch_size):
+
+    batch_actions = []
+    batch_outcome = []
+    tasks = []
+    # proposing in batch actions
+    for itr in range(i, min(i+batch_size, num_questions)):
+      if itr == num_questions - 1:
+        is_last_action = True
+
+      action = opt_obj.get_sample()
+      batch_actions.append(action)
+      tasks.append(
+          sight.create_task(
+              proposal.propose_actions(sight,
+                                      get_question_label_to_propose_actions(),
+                                      action_dict=action)))
+
+    print('batch_actions : ', batch_actions, len(batch_actions))
+    # wait for their output, update optimizer
+    batch_outcome = await asyncio.gather(*tasks)
+    print('batch_outcome : ', batch_outcome, len(batch_outcome))
+    vals = list(batch_actions)
+    # some mechanchism to calculate reward from the response of WS worker
+    reward = 100 #static
+
+    for b in range(len(batch_outcome)):
+      outcomes.append(batch_outcome[b])
+      rewards.append(reward)
+      actions.append(batch_actions[b])
+
+      #some way to document all the actions with its outcome
+      opt_obj.document_sample(batch_actions[b], reward, batch_outcome[b])
+
+  # return all actions with its rewards, outcomes
+  return actions, rewards, outcomes
+
+
+def main(sight: Sight) -> Tuple[float, Dict[str, int]]:
+
+  opt_obj = BayesOptOptimizerClient(sight)
```

Comment:
I like the direction overall.

Please move BayesOptOptimizerClient to propose_parallelly

I like that you're using the wrapper from sight.widgets.decision.optimizers.bayes_opt_client since you've already done the work integrating these optimizers and you don't need to do it again

Reply:
How does BayesOptOptimizerClient know which action attributes to propose for and their valid ranges?

Once you move it to the call to propose_parallely() you'll get those from the the decision_point() call and will be able to inform this instance of BayesOptimizerClient